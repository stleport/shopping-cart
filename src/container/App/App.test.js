import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import ConnectedApp, { App } from './App';

function AppSetup(newProps = {}) {
  const mockStore = configureMockStore();
  const store = mockStore({});
  const props = {};

  const AppJSX = (
    <Provider store={store}>
      <ConnectedApp />
    </Provider>
  );
  const AppComponent = shallow(AppJSX);
  return {
    props,
    newProps,
    AppComponent,
  };
}

it('renders correctly App component', () => {
  const { AppJSX } = AppSetup();
  const AppComponent = renderer.create(AppJSX).toJSON();
  expect(AppComponent).toMatchSnapshot();
});
