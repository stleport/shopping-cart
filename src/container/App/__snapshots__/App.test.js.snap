// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly App component 1`] = `null`;
